arkeocli/close.go

```diff
@@ -0,0 +1,65 @@
+package arkeocli
+
+import (
+	"fmt"
+
+	"github.com/arkeonetwork/arkeo/x/arkeo/types"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	"github.com/spf13/cobra"
+)
+
+func newCloseContractCmd() *cobra.Command {
+	closeContractCmd := &cobra.Command{
+		Use:   "close-contract",
+		Short: "close a contract",
+		Args:  cobra.ExactArgs(0),
+		RunE:  runCloseContractCmd,
+	}
+
+	flags.AddTxFlagsToCmd(closeContractCmd)
+	closeContractCmd.Flags().Uint64("contract-id", 0, "id of contract")
+	closeContractCmd.Flags().String("provider-pubkey", "", "provider pubkey")
+	closeContractCmd.Flags().String("client-pubkey", "", "client pubkey")
+	closeContractCmd.Flags().String("service", "", "service name")
+	return closeContractCmd
+}
+
+func runCloseContractCmd(cmd *cobra.Command, args []string) (err error) {
+	clientCtx, err := client.GetClientTxContext(cmd)
+	if err != nil {
+		return err
+	}
+
+	key, err := ensureKeys(cmd)
+	if err != nil {
+		return err
+	}
+	addr, err := key.GetAddress()
+	if err != nil {
+		return
+	}
+	clientCtx = clientCtx.WithFromName(key.Name).WithFromAddress(addr)
+	if err = client.SetCmdClientContext(cmd, clientCtx); err != nil {
+		return
+	}
+
+	contract, err := getContract(cmd)
+	if err != nil {
+		return err
+	}
+
+	if contract.Type == types.ContractType_PAY_AS_YOU_GO {
```

Comment:
you can remove this check, there are occasions when a client can close a PAY AS YOU GO, for instance when a provider has unbonded.